apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx

```diff
@@ -286,34 +298,89 @@ export class LibraryManagerDialog extends React.Component {
     });
   };
 
-  viewCode = library => {
+  viewCode = (library, mode) => {
     if (!library) {
       return;
     }
 
-    this.setState({viewingLibrary: library, isViewingCode: true});
+    this.setState({
+      displayLibrary: library,
+      displayLibraryMode: mode || DisplayLibraryMode.VIEW
+    });
   };
 
   closeLibraryManager = () => {
     this.setState({cachedClassLibraries: []});
     this.props.onClose();
   };
 
+  renderDisplayLibrary = () => {
+    const {displayLibrary, displayLibraryMode} = this.state;
+    if (
+      !displayLibrary ||
+      !Object.values(DisplayLibraryMode).includes(displayLibraryMode)
+    ) {
+      return;
+    }
+
+    const onClose = () =>
+      this.setState({
+        displayLibrary: null,
+        displayLibraryMode: DisplayLibraryMode.NONE
+      });
+
+    if (displayLibraryMode === DisplayLibraryMode.VIEW) {
+      return (
+        <LibraryViewCode
+          title={displayLibrary.name}
+          description={displayLibrary.description}
+          onClose={onClose}
+          sourceCode={displayLibrary.source}
+        />
+      );
+    }
+
+    if (displayLibraryMode === DisplayLibraryMode.UPDATE) {
+      return (
+        <LibraryViewCode
+          title={i18n.updateLibraryConfirmation({
+            libraryName: displayLibrary.name
+          })}
+          description={displayLibrary.description}
+          onClose={onClose}
+          sourceCode={displayLibrary.source}
+          buttons={
+            <div style={styles.updateButtons}>
+              <Button
+                text={i18n.cancel()}
+                color={Button.ButtonColor.gray}
+                onClick={onClose}
+              />
+              <Button
+                text={i18n.update()}
+                onClick={() => this.updateLibraryInProject(displayLibrary)}
+              />
+            </div>
+          }
+        />
+      );
+    }
+  };
+
   render() {
     const {isOpen} = this.props;
-    const {
-      isViewingCode,
-      importLibraryId,
-      viewingLibrary,
-      isLoading,
-      error
-    } = this.state;
+    const {importLibraryId, displayLibrary, isLoading, error} = this.state;
+
+    if (!isOpen) {
+      return null;
+    }
+
     return (
       <div>
```

Comment:
is this wrapping div now unnecessary?

Reply:
no, there are still 2 elements within this div (`<BaseDialog/>` and `<LibraryViewCode/>`)

Reply:
ah nevermind, sorry!